imbi/endpoints/components/handlers.py

```diff
@@ -101,11 +101,24 @@ async def delete(self, *args, **kwargs):
 
     @base.require_permission('admin')
     async def patch(self, *args, **kwargs):
-        await super().patch(*args, **kwargs)
+        result = await self.postgres_execute(
+            self.GET_SQL, {'package_url': kwargs['package_url']},
+            f'get-{self.NAME}')
+        if not result:
+            raise errors.ItemNotFound(instance=self.request.uri)
+
+        original = models.Component.model_validate(result.row)
+        original_dict = original.model_dump()
+        patch = jsonpatch.JsonPatch(self.get_request_body())
```

Comment:
Should probably catch `jsonpatch.InvalidJsonPatch` here

Reply:
I decided to add `errors.BadJsonPatch` and the more generic `errors.UnprocessableEntity` since we have similar code strewn about.